infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfigurator.java

```diff
@@ -0,0 +1,252 @@
+package org.corfudb.infrastructure.configuration;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.BatchProcessor;
+import org.corfudb.infrastructure.LogUnitServer;
+import org.corfudb.infrastructure.LogUnitServer.LogUnitParameters;
+import org.corfudb.infrastructure.LogUnitServerCache;
+import org.corfudb.infrastructure.ManagementAgent;
+import org.corfudb.infrastructure.ManagementServer;
+import org.corfudb.infrastructure.ServerContext;
+import org.corfudb.infrastructure.ServerThreadFactory;
+import org.corfudb.infrastructure.log.InMemoryStreamLog;
+import org.corfudb.infrastructure.log.StreamLog;
+import org.corfudb.infrastructure.log.StreamLogCompaction;
+import org.corfudb.infrastructure.log.StreamLogFiles;
+import org.corfudb.infrastructure.management.ClusterStateContext;
+import org.corfudb.infrastructure.management.FailureDetector;
+import org.corfudb.infrastructure.orchestrator.Orchestrator;
+import org.corfudb.protocols.wireprotocol.ClusterState;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuRuntime.CorfuRuntimeParameters;
+import org.corfudb.runtime.exceptions.UnreachableClusterException;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.Utils;
+import org.corfudb.util.concurrent.SingletonResource;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.corfudb.infrastructure.management.ClusterStateContext.HeartbeatCounter;
+
+/**
+ * This class is used to configure and create the corfu servers that share the same resources.
+ */
+@Slf4j
+public class ServerConfigurator {
+    // Global server context
+    @Getter
+    private final ServerContext context;
+
+    // Log Unit related configuration
+    private final long streamCompactionInitialDelay = 10L;
+    private final long streamCompactionPeriod = 45L;
+    private final TimeUnit streamCompactionUnits = TimeUnit.MINUTES;
+    private StreamLog streamLog;
+    private LogUnitParameters params;
+    private BatchProcessor batchProcessor;
+    private LogUnitServerCache logUnitServerCache;
+    private StreamLogCompaction streamLogCompaction;
+    private LogUnitServer logUnitServer;
+
+    // Management Server related configuration
+
+    /**
+     * The number of tries to be made to execute any RPC request before the runtime gives up and
+     * invokes the systemDownHandler.
+     * This is set to 60  based on the fact that the sleep duration between RPC retries is
+     * defaulted to 1 second in the Runtime parameters. This gives the Runtime a total of 1 minute
+     * to make progress. Else the ongoing task is aborted.
+     */
+    private final int systemDownHandlerTriggerLimit = 60;
```

Comment:
Why is 60 better than 20 (i.e. default value)?

Reply:
That's the default for a management server
https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java#L92